### PR TITLE
Replace #[doc] error comments with /// type

### PR DESCRIPTION
**Pull request by KnoxSamuel**
Fixed issue #4313, 12 comments changed

## Changes
This commit updates error doc comments with /** */.
It also moves or removes extraneous skip attributes.

    modified:   src/cpu-template-helper/src/template/verify/mod.rs
    modified:   src/vmm/src/cpu_config/x86_64/cpuid/mod.rs        
    modified:   src/vmm/src/cpu_config/x86_64/cpuid/normalize.rs  
    modified:   src/vmm/src/devices/virtio/net/tap.rs
    modified:   src/vmm/src/mmds/mod.rs
    modified:   src/vmm/src/mmds/token.rs
    modified:   src/vmm/src/persist.rs
    modified:   src/vmm/src/vmm_config/machine_config.rs
    modified:   src/vmm/src/vmm_config/mmds.rs

## Reason

Issue #4313 requested changes to error comment types across the codebase.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [X] If a specific issue led to this PR, this PR closes the issue.
- [X] The description of changes is clear and encompassing.
- [X] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Comment by KnoxSamuel** [comment]
I was trying to fix my author name (I only had my GitHub username KnoxSamuel set on my profile without a actual name set). But when I did an amend and rebase, it added my sign-off to Johnathan's commit. I then rebased again to fix his commit msg, and now the history says 'JonathanWoollett-Light
 authored and
 KnoxSamuel
 committed
' ... is this an issue?

File: src/cpu-template-helper/src/template/verify/mod.rs [hunk 1]
```diff
@@ -23,7 +23,7 @@ pub enum VerifyError {
     /// {0} not found in CPU configuration.
     KeyNotFound(String),
     #[rustfmt::skip]
-    #[doc = "Value for {0} mismatched.\n{1}"]
+    /** Value for {0} mismatched.\n{1} */
```
Comment (roypat) [inline src/cpu-template-helper/src/template/verify/mod.rs:26]:
```suggestion
    /** Value for {0} mismatched.\
    {1} */
```

File: src/vmm/src/persist.rs [hunk 2]
```diff
@@ -158,7 +158,7 @@ pub enum CreateSnapshotError {
     /// Cannot get dirty bitmap: {0}
     DirtyBitmap(VmmError),
     #[rustfmt::skip]
-    #[doc = "The virtio devices use a features that is incompatible with older versions of Firecracker: {0}"]
+    /** The virtio devices use a features that is incompatible with older versions of Firecracker: {0} */
```
Comment (roypat) [inline src/vmm/src/persist.rs:161]:
```suggestion
    /** The virtio devices use a features that is incompatible with older\
    versions of Firecracker: {0} */
```

**Review by roypat — COMMENTED** [review]
Hi @KnoxSamuel, thanks for tackling this one! 
I think it would be nice if we could wrap the comments, part of the advantage of using the block-doc comments is that we no longer need to have everything on a single line. I've included one example of what I mean as a review comment. Also, could we hoist all the `#[rustfmt::skip]` directives to the top of the enums?

**Comment by roypat** [comment]
> I was trying to fix my author name (I only had my GitHub username KnoxSamuel set on my profile without a actual name set). But when I did an amend and rebase, it added my sign-off to Johnathan's commit. I then rebased again to fix his commit msg, and now the history says 'JonathanWoollett-Light authored and KnoxSamuel committed' ... is this an issue?

Mhh, I dont think Jonathan's commit should be a part of this PR at all. Can you drop it and rebase against main again? :o

**Comment by KnoxSamuel** [comment]
Thanks for the feedback. @roypat 

I've fixed my fork's history, moved the skip attributes & wrapped those comments to col 75 or less. Does everything look good now? :)

**Review by ShadowCurse — COMMENTED** [review]
It seems the `\` at the end is preserved in the comments. Also making comments multi-line makes them to have a `\n` and this breaks some tests.
I think the simplest things we can do are:
- make comments one line again
- use `///` for them (because if the are one line why use `//* */`)
- leave `#[rustfmt::skip]` at the top of structs to prevent long lines being reformatted into multi-lines.

**Comment by KnoxSamuel** [comment]
Okay no problem. 
1. Should I rebase my commit or push a new commit to my branch?
2. Undo `#[rustfmt::skip]` changes where they were before, right above the structs/variants?

**Comment by ShadowCurse** [comment]
> Should I rebase my commit or push a new commit to my branch?

I think you can rebase. Keep all changes in one commit in this case is fine.

> Undo #[rustfmt::skip] changes where they were before, right above the structs/variants?

You can keep them above the structs, so same way they are currently in the PR.

**Comment by KnoxSamuel** [comment]
Let me know if there's anything I missed now!

File: src/cpu-template-helper/src/template/verify/mod.rs [hunk 3]
```diff
@@ -22,8 +22,7 @@ pub use x86_64::verify;
 pub enum VerifyError {
     /// {0} not found in CPU configuration.
     KeyNotFound(String),
-    #[rustfmt::skip]
-    #[doc = "Value for {0} mismatched.\n{1}"]
+    /// Value for {0} mismatched.\n{1}
```
Comment (ShadowCurse) [inline src/cpu-template-helper/src/template/verify/mod.rs:25]: Ok, the last thing. As it appears `#[doc = "Value for {0} mismatched.\n{1}"]` and `/// Value for {0} mismatched.\n{1}` are not equal. The later converts `\n` into `\\n`. So in this case let's use the `/** */`. Never though Rust comments with new lines will be such a pain.
```diff
-    /// Value for {0} mismatched.\n{1}
+    /** Value for {0} mismatched.
+    {1} */
```

Comment (KnoxSamuel) [reply src/cpu-template-helper/src/template/verify/mod.rs:25]: Haha strange. Thanks for the help.

**Review by roypat — APPROVED** [review]

**Review by ShadowCurse — APPROVED** [review]

**Pull request merged**